momepy/functional/_dimension.py

```diff
@@ -0,0 +1,144 @@
+import numpy as np
+import pandas as pd
+import shapely
+
+__all__ = [
+    "volume",
+    "floor_area",
+    "courtyard_area",
+    "longest_axis_length",
+    "perimeter_wall",
+]
+
+
+def volume(area, height):
+    """
+    Calculates volume of each object in given GeoDataFrame based on its height and area.
+
+    .. math::
+        area * height
+
+    Parameters
+    ----------
+    area : array_like
+        array of areas
+    height : array_like
+        array of heights
+
+    Returns
+    -------
+    array_like
+    """
+    return area * height
+
+
+def floor_area(area, height, floor_height=3):
+    """Calculates floor area of each object based on height and area.
+
+    The number of
+    floors is simplified into the formula: ``height // floor_height``. B
+    y default one floor is approximated to 3 metres.
+
+    .. math::
+        area * \\frac{height}{floor_height}
+
+
+    Parameters
+    ----------
+    area : array_like
+        arrat of areas
+    height : array_like
+        array of heights
+    floor_height : float | array_like, optional
+        float denoting the uniform floor height or an array_like reflecting the building
+        height by geometry, by default 3
+
+    Returns
+    -------
+    array_like
+    """
+    return area * (height // floor_height)
+
+
+def courtyard_area(gdf):
+    """Calculates area of holes within geometry - area of courtyards.
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+        A GeoDataFrame containing objects to analyse.
+
+    Returns
+    -------
+    pandas.Series
+    """
+    return pd.Series(
+        shapely.area(shapely.polygons(shapely.get_exterior_ring(gdf.geometry.array)))
+        - gdf.area,
+        index=gdf.index,
+        name="courtyard_area",
+    )
+
+
+def longest_axis_length(gdf):
+    """Calculates the length of the longest axis of object.
+
+    Axis is defined as a
+    diameter of minimal bounding circle around the geometry. It does
+    not have to be fully inside an object.
+
+    .. math::
+        \\max \\left\\{d_{1}, d_{2}, \\ldots, d_{n}\\right\\}
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+        A GeoDataFrame containing objects to analyse.
+
+    Returns
+    -------
+    pandas.Series
+    """
+    return shapely.minimum_bounding_radius(gdf.geometry) * 2
+
+
+def perimeter_wall(gdf, graph=None):
+    """
+    Calculate the perimeter wall length the joined structure.
```

Comment:
Is this the correct docstring? Should it read more like [`Perimeter`](https://github.com/pysal/momepy/blob/2051f4b7aaa353bf28c12f817c2f6f8c7f628ae5/momepy/dimension.py#L65)?

Reply:
Yes, it is correct. It is reimplementation of [`PerimeterWall`](https://github.com/pysal/momepy/blob/2051f4b7aaa353bf28c12f817c2f6f8c7f628ae5/momepy/dimension.py#L845) not `Perimeter`. `Perimeter` will be removed and users pointed to `.length`.